### PR TITLE
test: add branch_stage property tests

### PR DESCRIPTION
**Pull request by gabriele-0201**

### CODE REVIEW

**Comment by gabriele-0201** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/508?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#508** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/508?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* **#507** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/507?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#506** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/506?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#505** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/505?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#504** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/504?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#503** <a href="https://app.graphite.dev/github/pr/thrumdev/nomt/503?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gabriele-0201 and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: nomt/src/beatree/index.rs [hunk 1]
```diff
@@ -44,6 +44,13 @@ impl Index {
     pub fn insert(&mut self, separator: Key, branch: Arc<BranchNode>) -> Option<Arc<BranchNode>> {
         self.first_key_map.insert(separator, branch)
     }
+
+    #[cfg(test)]
+    pub fn new() -> Self {
```
Comment (rphmeier) [inline nomt/src/beatree/index.rs:49]: This is unnecessary, because `Index::default` exists

File: nomt/src/beatree/ops/update/branch_updater.rs [hunk 2]
```diff
@@ -468,7 +468,7 @@ impl BranchBulkSplitter {
 }
 
 #[cfg(test)]
-mod tests {
+pub mod tests {
```
Comment (rphmeier) [inline nomt/src/beatree/ops/update/branch_updater.rs:471]: I'll recommend a small refactor here: extract the relevant code to a `test_helpers` module.

It's a little rough to expose `mod tests` publicly. Not a huge deal, but not great style.

**Review by rphmeier — APPROVED** [review]

**Comment by rphmeier** [comment]
### Merge activity

* **Oct 24, 6:32 PM EDT**: A user started a stack merge that includes this pull request via [Graphite](https://app.graphite.dev/github/pr/thrumdev/nomt/508).

**Pull request merged**